### PR TITLE
Properly validate MachineAutoscaler target references

### PR DESCRIPTION
**Pull request by bison**
This adds validation to MachineAutoscaler target references.
Previously, if for example a reference had no name set, the dynamic
client would return a list and cause a panic.  This prevents fetching
invalid or unsupported target references.

### CODE REVIEW

File: pkg/controller/machineautoscaler/machineautoscaler_controller.go [hunk 1]
```diff
@@ -228,8 +234,8 @@ func (r *Reconciler) GetTarget(ref *corev1.ObjectReference) (*MachineTarget, err
 	obj := &unstructured.Unstructured{}
 	gvk := ref.GroupVersionKind()
 
-	if !SupportedTarget(gvk) {
-		return nil, ErrUnsupportedTarget
+	if valid, err := ValidateReference(ref); !valid || err != nil {
```
Comment (frobware) [inline pkg/controller/machineautoscaler/machineautoscaler_controller.go]: Is it more prudent to first look at `err`, then `!valid`.

Comment (bison) [reply pkg/controller/machineautoscaler/machineautoscaler_controller.go]: Yeah, this is a little weird. I've updated the comment for the validation function to reflect the assumptions about returning non-nil errors. I also updated the test to ensure we always return a non-nil error for invalid references.
